docker/deployment/Dockerfile

```diff
@@ -0,0 +1,50 @@
+# STAGE: Building the server
+FROM maven:3.6.3-jdk-13 AS server-build
+RUN mkdir /metadatahub
+WORKDIR /metadatahub
+COPY server server
+COPY web-ui web-ui
+WORKDIR /metadatahub/server
+RUN mvn -f pom.xml clean package
+
+# STAGE: FINAL IMAGE
+FROM ubuntu:latest as app
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive \
+    METADATAHUB_ENV=/metadatahub/configs/environment.deployment.json \
+    SERVER_CONFIG=/metadatahub/configs/deployment.env
+# Install system dependencies
+RUN apt update -y && \
+    apt install -y \
+        tzdata \
+        libpq-dev \
+        python3.8 \
+        python3-pip \
+        openjdk-13-jre \
+        postgresql \
+        postgresql-contrib
+RUN mkdir /metadatahub
+WORKDIR /metadatahub
+# Setup database
+COPY database database
+COPY docker/deployment/scripts scripts
+USER postgres
+RUN /etc/init.d/postgresql start && \
+        psql -f database/metadatahub-role.sql && \
+        psql -f database/metadatahub-database.sql && \
+        psql metadatahub -f database/metadatahub-schemata.sql && \
+        psql metadatahub -f database/metadatahub-data.sql && \
+    scripts/database.sh
+USER root
+# Install crawler dependencies
+COPY crawler/requirements crawler/requirements
+RUN pip3 install -r crawler/requirements/base.txt
+COPY crawler crawler
+# Copy server artifacts
+COPY --from=server-build metadatahub/server/target/metadata-hub-server-application-fat.jar server/
+COPY --from=server-build metadatahub/web-ui web-ui
+# Copy configs, export ports and set entrypoint and default arguments via cmd
```

Comment:
we only need the ....fat.jar.
maven includes into this with the help from webpack all necassary files.

maven:
```
 <configuration>
                    <workingDirectory>../web-ui/</workingDirectory>
                    <nodeVersion>v12.16.3</nodeVersion>
                    <npmVersion>6.14.5</npmVersion>
 </configuration>
```
[...]
``` 
<execution>
                        <id>webpack-build</id>
                        <goals><goal>npm</goal></goals>
                        <configuration>
                            <arguments>run dev-build</arguments>
                        </configuration>
</execution>
```

in the package.json-file:
```
    "scripts": {
        "build": "webpack",
        "dev-build": "webpack -o ../server/src/main/resources/static-files/bundle.js",
        "runtime-build": "webpack -o ../server/target/classes/static-files/bundle.js",
       "test": "echo \"Error: no test specified\" && exit 1"
    },
```

so maven calls the dev-build (`<arguments>run dev-build</arguments>`). This bundle the files (there are more bundel.js which are then outputet , main.bundle.js, bundle.js, something like that). This goes to the ressource-directory. And in the final, if maven could compile the java-code, all of that goes to the jar-file. 
So its only necassary, that maven can access `  <workingDirectory>../web-ui/</workingDirectory>`

Maybe its a good idea to cache the **node/ and node_modules/**, so the build could be faster

Reply:
Thanks a lot, I'll work on that :)